File: src/main/kotlin/com/fugisawa/j2k/livecoding/java/Main.java
package com.fugisawa.j2k.livecoding.java;

import java.util.Collection;
import java.util.HashSet;

public class Main {

    public static void main(String[] args) {

        final Collection<Point> points = new HashSet<>() {{
            add(new Point(1, 1));
            add(new Point(2, 2));
            add(new Point(3, 3));
            add(new Point(4, 4));
            add(new Point(5, 5));
        }};

        final Circle circle = new Circle(new Point(2, 2), 1);

        final long count = circle.numberOfInscribedPoints(points);
        final Collection<Point> inscribedPoints = circle.inscribedPoints(points);
        final Point aRandomInscribedPoint = circle.aRandomInscribedPoint(points);

        System.out.println("Inscribed points: " + inscribedPoints);
        System.out.println("Number of inscribed points: " + count);

        if (aRandomInscribedPoint != null) {
            System.out.println("A random inscribed point: " + aRandomInscribedPoint);
            doSomethingWithPoint(aRandomInscribedPoint);
        } else {
            throw new IllegalStateException("aRandomInscribedPoint is null");
        }

    }

    public static void doSomethingWithPoint(Point point) {
        System.out.println("X is " + point.getX());
        System.out.println("Y is " + point.getY());
    }
}

File: src/main/kotlin/com/fugisawa/j2k/livecoding/kotlin/Main.kt
package com.fugisawa.j2k.livecoding.kotlin

fun main() {
    val points = setOf(
        1 x 1,
        2 x 2,
        3 x 3,
        4 x 4,
        5 x 5,
    )
    val circle = 2 x 2 radius 1

    val count = circle numberOfInscribed points
    val inscribedPoints = circle inscribed points
    val aRandomInscribedPoint = circle aRandomInscribedFrom points

    println("Inscribed points: $inscribedPoints")
    println("Number of inscribed points: $count")

    aRandomInscribedPoint
        ?.let {
            println("A random inscribed point: $it")
            doSomethingWithPoint(it)
        }
        ?: throw IllegalStateException("aRandomInscribedPoint is null")

} 

fun doSomethingWithPoint(point: Point) {
    println("X is " + point.x)
    println("Y is " + point.y)
}
